genai-perf/genai_perf/metrics/telemetry_statistics.py

```diff
@@ -56,21 +56,26 @@ def __init__(self, metrics: TelemetryMetrics):
                 self._stats_dict[attr][gpu_index]["avg"] = (
                     self._statistics._calculate_mean(gpu_data)
                 )
-            if not self._is_constant_metric(attr):
-                if gpu_data is None or gpu_index is None:
-                    continue
-
-                percentile_results = self._statistics._calculate_percentiles(gpu_data)
-                for percentile_label, percentile_value in percentile_results.items():
-                    self._stats_dict[attr][gpu_index][
-                        percentile_label
-                    ] = percentile_value
-                min, max = self._statistics._calculate_minmax(gpu_data)
-                self._stats_dict[attr][gpu_index]["min"] = min
-                self._stats_dict[attr][gpu_index]["max"] = max
-                self._stats_dict[attr][gpu_index]["std"] = (
-                    self._statistics._calculate_std(gpu_data)
-                )
+                if not self._is_constant_metric(attr):
```

Comment:
Fixing this. It was meant to be inside the loop but was outside the loop, so only the last iteration of Telemetry metrics were captured.

Reply:
Thats a nefarious little bug.
I feel like robust testing should catch this type of bug.